src/Redux/Book/reducer.js

```diff
@@ -1,13 +1,20 @@
 const ADD_BOOKS = 'bookstore/books/ADD_BOOKS';
+const FILTER_BOOKS = 'bookstore/books/FILTER_BOOKS';
 
 export const addBook = (payload) => ({
   type: ADD_BOOKS,
   payload,
 });
-
+export const filterBooks = (payload) => ({
+  type: FILTER_BOOKS,
+  payload,
+});
 const initialState = [];
+
 const bookReducer = (state = initialState, action) => {
   switch (action.type) {
+    case FILTER_BOOKS:
+      return { ...state, filterBooks: action.filter };
```

Comment:
- [ ] Filter_books is actually removing a book and it should be using the filter method, kindly change it so it works as expected. your code may be like `return state.filter(book => book.id !== id)`